src/IO/ReadBufferFromPocoSocket.cpp

```diff
@@ -28,10 +28,23 @@ bool ReadBufferFromPocoSocket::nextImpl()
     ssize_t bytes_read = 0;
     Stopwatch watch;
 
+    int flags = 0;
+    if (async_callback)
+        flags |= MSG_DONTWAIT;
```

Comment:
Note that `async_socket_for_remote` is completely broken with unbundled poco, since it cannot handle `MSG_DONTWAIT` correctly.

But IIUC arcadia build does not support distributed queries, so it is not a problem there?

Reply:
Indeed.

@azat does it cause any problem for you? I think I can just read form socket manually, avoid using poco in this case.

Reply:
>@azat does it cause any problem for you?

For development I'm using unbundled build (since it is faster), and I was using unbundled poco there, but now I've switched to bundled poco there so it is not a problem anymore

>I think I can just read form socket manually, avoid using poco in this case.

Maybe it is worth it, since this patches already has some low-level code anyway (epoll_ctl and similar), although I'm not sure.